scripts/validate-artifacts.ps1

```
@@ -106,39 +106,39 @@ foreach ($requiredSubnet in $requiredSubnets) {
-    throw "Unable to verify webservers NSG. Please make sure that it has only 1 rule, which allows inbound traffic on TCP ports 80 and 443 from any source port and IP address and try again."
-}
- 
+# if ($webRule) {
+#     Write-Output "`u{2705} Checked if webservers NSG allows HTTP traffic - OK."
+# } else {
+#     Write-Output `u{1F914}
+#     throw "Unable to verify webservers NSG. Please make sure that it has only 1 rule, which allows inbound traffic on TCP ports 80 and 443 from any source port and IP address and try again."
+# }
+
 $managementNSG = ( $TemplateObject.resources | Where-Object -Property type -EQ "Microsoft.Network/networkSecurityGroups" | Where-Object {$_.name.Contains("management")})
-if ($managementNSG.properties.securityRules.Count -eq 1) { 
+if ($managementNSG.properties.securityRules.Count -eq 2) {
     Write-Output "`u{2705} Checked if management NSG has only 1 rule - OK."
-} else { 
+} else {
     Write-Output `u{1F914}
     throw "Unable to verify management NSG. Please make sure that it has only 1 rule and try again."
 }
 $sshRule = $managementNSG.properties.securityRules | Where-Object {$_.properties.sourcePortRange -eq "*" -and $_.properties.access -eq "Allow" -and $_.properties.protocol -eq "TCP" -and $_.properties.direction -eq "Inbound" -and $_.properties.destinationPortRange -eq "22"}
-if ($sshRule) { 
+if ($sshRule) {
     Write-Output "`u{2705} Checked if management NSG allows SSH traffic - OK."
-} else { 
+} else {
     Write-Output `u{1F914}
     throw "Unable to verify management NSG. Please make sure that it has only 1 rule, which allows inbound traffic on TCP port 22 from any source port and IP address and try again."
 }
 
 $databaseNSG = ( $TemplateObject.resources | Where-Object -Property type -EQ "Microsoft.Network/networkSecurityGroups" | Where-Object {$_.name.Contains("database")})
-if ($databaseNSG.properties.securityRules.Count -eq 0) { 
+if ($databaseNSG.properties.securityRules.Count -eq 1) {
```

The database NSG check asserts `Count -eq 1` (which corresponds to the Allow-VNet rule), but the success message says "Checked if database NSG has no rules - OK." This is contradictory. Update the message to reflect the actual expectation (1 rule: Allow-VNet) or change the test if the intended expectation differs. The current mismatch is confusing and may hide failures.